File: sample/dracula.dart
class Vampire {
  const Vampire({
    required this.location,
    required this.birthDate,
    required this.deathDate,
    required this.weaknesses,
  });

  final String location;
  final int birthDate;
  final int deathDate;
  final List<String> weaknesses;

  int get age => deathDate - birthDate;
}

void main() {
  const dracula = Vampire(
    location: "Transylvania",
    birthDate: 1428,
    deathDate: 1476,
    weaknesses: ["Sunlight", "Garlic"],
  );

  const alucard = Vampire(
    location: "Wallachia",
    birthDate: 1458,
    deathDate: 1510,
    weaknesses: ["Holy Water"],
  );

  print("Dracula: ${dracula.age}, Alucard: ${alucard.age}");
}

File: sample/dracula.ex
defmodule Vampire do
  @enforce_keys [:location, :birth_date, :death_date, :weaknesses]
  defstruct [:location, :birth_date, :death_date, :weaknesses]

  def age(%__MODULE__{} = vampire), do: vampire.death_date - vampire.birth_date
end

dracula = struct!(Vampire,
  location: "Transylvania",
  birth_date: 1428,
  death_date: 1476,
  weaknesses: ["Sunlight", "Garlic"]
)

alucard = struct!(Vampire,
  location: "Wallachia",
  birth_date: 1458,
  death_date: 1510,
  weaknesses: ["Holy Water"]
)

IO.inspect(%{dracula_age: Vampire.age(dracula), alucard_age: Vampire.age(alucard)})
